File: Python/Level1/Level1_문자열 내림차순으로 배치하기.py
def solution(s):
    answer = ""
    l = list(s)

    l.sort()
    l.reverse()

    for i in range(0, len(l)) :
        if(l[i].islower()) :
            front = l[i:]
            back = l[:i]
            break

    for i in front :
        answer += i

    for i in back :
        answer += i

    return answer

print(solution("caCbdA"))
print(solution("Zbcdefg"))


"""
# Best Answer
def solution(s):
    return "".join(sorted(list(s), reverse=True));
"""

File: C++/Level 1/Level1_문자열 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(string s) {
    string answer = "";
    vector<char> sTmp;

    for (int i = 0; i < s.length(); i++) {
        sTmp.push_back(s[i]);
    }

    sort(sTmp.begin(), sTmp.end(), greater<char>());

    for (auto i : sTmp) {
        answer += i;
    }

    return answer;
}

/* Best Answer
 string solution(string s) {
    sort (s.begin(), s.end(), greater<char>());
    return s;
}
*/


int main() {
    
    string s = "Zbcdefg";

    cout << solution(s);

    return 0;
}